File: tools/dbench/answers/go_haiku/s3/e3_validate.go
package main

import (
	"fmt"
)

func main() {
	ages := []int{34, -2, 19, 150, 42, 27}
	validCount := 0
	invalidCount := 0
	validSum := 0

	for _, age := range ages {
		if age >= 0 && age <= 120 {
			validCount++
			validSum += age
		} else {
			invalidCount++
		}
	}

	fmt.Println(validCount)
	fmt.Println(invalidCount)
	fmt.Println(validSum)
}

File: tools/dbench/answers/rust_haiku/s1/e3_validate.rs
fn main() {
    let ages = [34, -2, 19, 150, 42, 27];
    let mut valid_count = 0;
    let mut invalid_count = 0;
    let mut valid_sum = 0;

    for &age in &ages {
        if age >= 0 && age <= 120 {
            valid_count += 1;
            valid_sum += age;
        } else {
            invalid_count += 1;
        }
    }

    println!("{}", valid_count);
    println!("{}", invalid_count);
    println!("{}", valid_sum);
}
